ClassLibrary/ClassLibrary/DepartmentReaders.cs

```diff
@@ -40,12 +39,25 @@ public Reader GetReader(int IDReader)
 
         public bool CheckBorrowedBook(Book book)
         {
+            if (book == null)
+            {
+                throw new ArgumentNullException("Null Argument \"book\" in AddBook");
+            }
             Book borrowedBook = Readers.SelectMany(x => x.BorrowedBooks).FirstOrDefault(x=>x==book);
             return borrowedBook != null;
         }
 
+        public int CountBorrowedBooksWithISBN(String ISBN)
+        {
+            return Readers.SelectMany(x => x.BorrowedBooks.Where(y => y.ISBN == ISBN)).Count();
```

Comment:
Please don't change this code! 
just think (and keep results in mind, we'll address it later) what would be the performance of this call for a Library with thousands of readers and a million books.

Reply:
I agree with you, 
I didn’t get an idea, maybe I don’t know, or I fell out (because I rarely use)
  
How would you do better ?

Reply:
we'll do it better later! for now, I just want you to be aware of performance. also, I hope that during development we will see that a properly designed (meaning easy to understand, implement and maintain) OOP solution gives performance improvement just as a side effect.